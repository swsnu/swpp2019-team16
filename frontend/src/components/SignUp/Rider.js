import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Box from '../common/Box/Box';
import Button from '../common/Button/Button';
import Typography from '@material-ui/core/Typography';

const RiderBlock = styled.div``;

Rider.propTypes = {
  form: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
};

function Rider({ form, onChange, onClick }) {
  return (
    <RiderBlock>
      <Box>
        <Typography variant="h4">Email</Typography>
        <input
          type="text"
          id="email-input"
          name="email"
          onChange={onChange}
          value={form.email}
        />
        <Typography variant="h4">Password</Typography>
        <input
          type="password"
          id="password-input"
          name="password"
          onChange={onChange}
          value={form.password}
        />
        <Typography variant="h4">Password Confirm</Typography>
        <input
          type="password"
          id="password-confirmation-input"
          name="passwordConfirmation"
          onChange={onChange}
          value={form.passwordConfirmation}
        />
        <Box>
          <Button
            children="Confirm"
            onClick={onClick}
            variant="contained"
            fullwidth="false"
          />
        </Box>
      </Box>
    </RiderBlock>
  );
}

export default Rider;
