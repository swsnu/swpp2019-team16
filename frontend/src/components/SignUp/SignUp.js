import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Box from '../common/Box/Box';
import Button from '../common/Button/Button';

const SignUpBlock = styled.div``;

SignUp.propTypes = {
  form: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
};

function SignUp({ form, onChange, onClick }) {
  return (
    <SignUpBlock>
      <h3>Join Ya-Ta!</h3>
      <h1>Create Your Account</h1>
      <Box>
        <h3>Email</h3>
        <input
          type="text"
          id="email-input"
          name="email"
          onChange={onChange}
          value={form.email}
        />
        <h3>Password</h3>
        <input
          type="password"
          id="password-input"
          name="password"
          onChange={onChange}
          value={form.password}
        />
        <h3>Password Confirm</h3>
        <input
          type="password"
          id="password-confirmation-input"
          name="passwordConfirmation"
          onChange={onChange}
          value={form.passwordConfirmation}
        />
        <h3>Vehicle Info (for Driver users)</h3>
        <input
          type="text"
          id="vehicle-info-input"
          name="vehicleInfo"
          onChange={onChange}
          value={form.vehicleInfo}
        />
        <h3>
          <Button
            children="Confirm"
            onClick={onClick}
            variant="contained"
            fullwidth="false"
          />
        </h3>
      </Box>
    </SignUpBlock>
  );
}

export default SignUp;
