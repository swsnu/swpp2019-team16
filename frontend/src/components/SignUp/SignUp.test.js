import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import SignUp from './SignUp';

describe('<SignUp />', () => {
  it('SHOULD match with snapshot WHEN all data in', async () => {
    const form = { email: 'dkim94', password: '123', vehicleInfo: '' };

    const { container } = render(
      <SignUp form={form} onChange={() => {}} onClick={() => {}} />,
    );
    expect(container).toMatchSnapshot();
  });
});
