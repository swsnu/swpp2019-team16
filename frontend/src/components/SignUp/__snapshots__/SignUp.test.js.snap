// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> SHOULD match with snapshot WHEN all data in 1`] = `
<div>
  <div
    class="sc-bdVaJa bDWFJH"
  >
    <h3>
      Join Ya-Ta!
    </h3>
    <h1>
      Create Your Account
    </h1>
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <h3>
        Email
      </h3>
      <input
        id="email-input"
        name="email"
        type="text"
        value="dkim94"
      />
      <h3>
        Password
      </h3>
      <input
        id="password-input"
        name="password"
        type="password"
        value="123"
      />
      <h3>
        Password Confirm
      </h3>
      <input
        id="password-confirmation-input"
        name="passwordConfirmation"
        type="password"
        value=""
      />
      <h3>
        Vehicle Info (for Driver users)
      </h3>
      <input
        id="vehicle-info-input"
        name="vehicleInfo"
        type="text"
        value=""
      />
      <h3>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-contained"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Confirm
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </h3>
    </div>
  </div>
</div>
`;
